File: CRPL.Data/Core/StructuredQuery.cs
namespace CRPL.Data;

public class StructuredQuery
{
    public string? Keyword { get; set; }
    public Sortable? SortBy { get; set; }
    public Dictionary<WorkFilter, string>? WorkFilters { get; set; }

    public override string ToString()
    {
        var msg = $"\nKeyword -> {Keyword}\nSorted By -> {SortBy.ToString()}\n";
        if (WorkFilters?.Keys != null)
            foreach (var filter in WorkFilters?.Keys)
            {
                msg += $"{filter.ToString()} -> {WorkFilters[filter]}";
            }

        return msg;
    }
}

public enum WorkFilter
{
    RegisteredAfter,
    RegisteredBefore,
}

public enum Sortable
{
    Created,
    Title,
}

File: CRPL.Web/ClientApp/src/app/_Models/StructuredQuery/StructuredQuery.ts
export interface StructuredQuery
{
  Keyword?: string;
  SortBy?: Sortable;
  WorkFilters?: Record<WorkFilter, string>;
}

export enum WorkFilter
{
  RegisteredAfter,
  RegisteredBefore,
  WorkType
}

export enum Sortable
{
  Created,
  Title,
  Registered
}
